src/Utilities/LockCheck.cs

```diff
@@ -12,10 +12,10 @@
 
 #nullable disable
 
-namespace Microsoft.Build.Tasks
+namespace Microsoft.Build.Utilities
 {
     [SupportedOSPlatform("windows")]
-    internal class LockCheck
+    public class LockCheck
```

Comment:
Since it is public now, all of the public classes, fields, properties, and members need XML doc comments describing why and how to use them. Can you add those?